### ISSUE TITLE
FQDN Pool_Members do not auto populate properly

### ISSUE THREAD

**Issue by ColinHeathman** [F5Networks/f5-appsvcs-extension#22]
##### ISSUE TYPE
 - Bug Report

##### AS3 BUILD/ VERSION
```
f5-appsvcs-3.2.0-7
```

##### BIGIP VERSION
```
  Product     BIG-IP
  Version     12.1.2
  Build       1.0.271
  Edition     Hotfix HF1
```

##### SUMMARY

FQDN Pool members do not auto populate IPs at the pool level. The IPs only populate at the node level.

FQDN Pools appear to be undocumented at the moment

##### STEPS TO REPRODUCE

Prerequisite: A DNS entry that returns a list of IPs. In this case it's **demo.development.svc.cluster.local**

**Example declaration:**

```
{
  "class": "ADC",
  "schemaVersion": "3.2.0",
  "id": "demo",
  "label": "Demo",
  "remark": "demo with FQDN pool",
  "demo_tenant": {
    "class": "Tenant",
    "demo_app": {
      "class": "Application",
      "template": "http",
      "serviceMain": {
        "class": "Service_HTTP",
        "virtualAddresses": [
          "192.168.0.21"
        ],
        "allowVlans": [
          "internal",
          "external"
        ],
        "pool": "demo_pool",
        "virtualPort": 80,
        "persistenceMethods": [
          "cookie"
        ],
        "profileHTTP": "basic",
        "layer4": "tcp",
        "profileTCP": "normal",
        "snat": "auto"
      },
      "demo_pool": {
        "class": "Pool",
        "members": [
          {
            "servicePort": 80,
            "addressDiscovery": "fqdn",
            "autoPopulate": true,
            "hostname": "demo.development.svc.cluster.local"
          }
        ]
      }
    },
    "defaultRouteDomain": 0
  }
}
```

##### EXPECTED RESULTS

**Node list**
```
FQDN
--
Address Type   | IPv4
Auto Populate  | Enabled
Interval       | Use TTL
Down Interval  | 5
```

**Pool member**
```
FQDN
--
Auto Populate | Enabled
```

##### ACTUAL RESULTS

**Node list**
```
FQDN
--
Address Type   | IPv4
Auto Populate  | Enabled
Interval       | Use TTL
Down Interval  | 5
```

**Pool member**
```
FQDN
--
Auto Populate | Disabled
```

**Comment by mdditt2000** [comment]
Hi Colin, I am investigating this issue. Can you please try changing the pool to the following

"members": [
{
"servicePort": 80,
"addressDiscovery": "fqdn",
"hostname": "webservers.fqdn.net",
"addressFamily": "IPv4",
"autoPopulate": false,
"queryInterval": 60,
"downInterval": 180
}
]

Thanks Mark

**Comment by ColinHeathman** [comment]
Here's the results from your test:


Node list ( Local Traffic  ››  Nodes : Node List  ››  demo.development.svc.cluster.local )

```
FQDN
--
Address Type   | IPv4
Auto Populate  | Disabled
Interval       | 60
Down Interval  | 180
```

Pool member ( Local Traffic  ››  Pools : Pool List  ››  demo_pool )

```
FQDN
--
Auto Populate | Disabled
```

Setting autoPopulate to 'false' only causes the first IP address in the DNS record to be added to the list.

ie. the only ephemeral node in the list is
`demo.development.svc.cluster.local-10.42.0.196`

but there should be 3

```
nslookup demo.development.svc.cluster.local

Name:      demo.development.svc.cluster.local
Address 1: 10.42.0.196
Address 2: 10.42.2.210
Address 3: 10.42.4.111
```

**Comment by mdditt2000** [comment]
Hi @ColinHeathman 

The FQDN address discovery defect will be resolved in 3.4. AS3.4 will be available early September

**Comment by mdditt2000** [comment]
@ColinHeathman AS3.4 was published today https://github.com/F5Networks/f5-appsvcs-extension/blob/master/dist/f5-appsvcs-3.4.0-2.noarch.rpm

**Comment by JosephPJordan** [comment]
Going to close this issue as this was fixed in AS3 3.4
